test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -2825,11 +2895,26 @@ def matcher_check_fn():
             )
 
 
-@dynamo_config.patch({"dynamic_shapes": True, "assume_static_by_default": False})
 class TestDynamicPatternMatcher(TestPatternMatcherBase):
+    def setUp(self):
```

Comment:
The origin decorator `@dynamo_config.patch({"dynamic_shapes": True, "assume_static_by_default": False})` will modify the base calss of `TestDynamicPatternMatcher` from `TestPatternMatcherBase` to `TestDynamicPatternMatcher`,  then cause the instantiate_device_type_tests failed.